fms/datasets/__init__.py

```diff
@@ -1,11 +1,16 @@
+from torch.utils.data import Dataset
+from typing import Callable, Mapping, Optional
 from fms.datasets.instructions import JsonInstructions
 from fms.datasets import text
 from fms.utils.tokenizers import BaseTokenizer
 
-__dataset_factory = {"instruction": JsonInstructions, "text": text.causaltext}
+__dataset_factory: Mapping[str, Callable[[str, BaseTokenizer], Dataset] | type] = {
+    "instruction": JsonInstructions,
+    "text": text.causaltext,
+}
 
 
-def get_dataset(name: str, tokenizer: BaseTokenizer, data: str = None, **kwargs):
+def get_dataset(name: str, tokenizer: BaseTokenizer, data: str = "", **kwargs):
```

Comment:
Should this not be an optional[str] = None

Reply:
The challenge with making it an `Optional` is that then for mypy to pass all of the functions have to be `Optional`s as well. Generally, functions getting a dataset do require you to pass in some path to the data, and a `None` path would be an error. Starting most dataset factories with `if path is None: raise...` seems more awkward than just using an empty string for cases with no path info.